js/src/offcanvas.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * --------------------------------------------------------------------------
+ * Bootstrap (v5.0.0-beta2): offcanvas.js
+ * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
+ * --------------------------------------------------------------------------
+ */
+
+import {
+  defineJQueryPlugin,
+  getElementFromSelector,
+  getSelectorFromElement,
+  getTransitionDurationFromElement,
+  isVisible
+} from './util/index'
+import { hide as scrollBarHide, reset as scrollBarReset } from './util/scrollbar'
+import Data from './dom/data'
+import EventHandler from './dom/event-handler'
+import BaseComponent from './base-component'
+import SelectorEngine from './dom/selector-engine'
+
+/**
+ * ------------------------------------------------------------------------
+ * Constants
+ * ------------------------------------------------------------------------
+ */
+
+const NAME = 'offcanvas'
+const DATA_KEY = 'bs.offcanvas'
+const EVENT_KEY = `.${DATA_KEY}`
+const DATA_API_KEY = '.data-api'
+const ESCAPE_KEY = 'Escape'
+const DATA_BODY_ACTIONS = 'data-bs-body'
```

Comment:
Better if you can accept this option in the configuration object (Like other plugins) :thinking:

Reply:
@Rohit. I really would prefer it to keep it simple right now, in order to proceed and to avoid more mess. 
If you find it so important, I can change it after. I am not very confident with the `Config` object, as every component uses it, in a different way